### PR TITLE
Add ability to specify generated time series file output path

### LINKED ISSUE #116: Need ability for user to select where generated time series are written
**Is your feature request related to a problem? Please describe.**

Currently the time series generation in CUPiD is hardcoded to write to the same directory path that the history files exist in.  However, there are situations (like the CESM tutorial) where this may not be possible, as a user may only have read access to the original model run or archive directory.

**Describe the solution you'd like**
Can fix this issue by adding a new, optional `ts_output_dir` config YAML variable to the `timeseries` config group, which if present will override the default time series path with a new path provided by the user.

**Describe alternatives you've considered**
I imagine a command line argument could provide the same functionality, but it looks like all of them are logicals, so I figured adding an optional YAML variable would be better.

**Additional context**
I will open a PR for this shortly.

### PR DESCRIPTION
**Pull request by nusbaume**
Fixes #116 

### All Submissions:

* [x] Have you followed the guidelines in our Contributer's Guide](https://github.com/NCAR/CUPiD/wiki/Contributor's-Guide) (including the `pre-commit` check)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by nusbaume** [comment]
@mnlevy1981 @TeaganKing please let me know what tests you want to have run with this branch as specified in the checklist above (I obviously tested it in my own config file).  Thanks!

**Comment by TeaganKing** [comment]
Hi @nusbaume , We don't have robust testing set up yet (that PR template checklist was a bit preemptive), so I think just testing with your own config file is great.

**Comment by TeaganKing** [comment]
It looks like the pre-commit is also passing.  Whenever it's ready, feel free to request a review.

**Comment by nusbaume** [comment]
@TeaganKing thanks for letting me know!  Sadly it looks like I am unable to add reviewers to this PR.

File: cupid/run.py [hunk 1]
```diff
@@ -110,6 +110,26 @@ def run(
...
+                    ),
+                ]
+
+                if "ts_output_dir" in timeseries_params:
+                    ts_output_dir = [
+                        os.path.join(
+                            timeseries_params["ts_output_dir"],
+                            f"{component}", "proc", "tseries",
+                        ),
+                    ]
```
Comment (mnlevy1981) [inline cupid/run.py:130]: One suggestion: can we keep `ts_output_dir` as a string, and pass `[ts_output_dir]` to `cupid.timeseries.create_time_series()`? (A separate issue ticket might be to update `create_time_series` to accept strings or lists as arguments, so list-ifying a handful of arguments seems silly... but that will be easier to address if `ts_output_dir` is a string.)

Also, a question: an alternate approach to this block of code would be

```py3
                if "ts_output_dir" in timeseries_params:
                    ts_output_dir =os.path.join(
                        timeseries_params["ts_output_dir"],
                        f"{component}", "proc", "tseries",
                    )
                else:
                    ts_output_dir = os.path.join(
                        global_params["CESM_output_dir"],
                        timeseries_params["case_name"],
                        f"{component}", "proc", "tseries",
                    )
```

Is there a reason to prefer one implementation over the other? If you want to keep

```py3
var=value
if condition:
    var=other_value
```

I see one `if-else` block in `run.py`, could you change that one?

```diff
        # Doing initial subsetting on full catalog, e.g. to only use certain cases

+        cat_path = full_cat_path
        if "subset" in control["data_sources"]:
            first_subset_kwargs = control["data_sources"]["subset"]
            cat_subset = full_cat.search(**first_subset_kwargs)
            # This pulls out the name of the catalog from the path
            cat_subset_name = full_cat_path.split("/")[-1].split(".")[0] + "_subset"
            cat_subset.serialize(
                directory=temp_data_path, name=cat_subset_name, catalog_type="file",
            )
            cat_path = temp_data_path + "/" + cat_subset_name + ".json"
-        else:
-            cat_path = full_cat_path
```

**Review by mnlevy1981 — CHANGES_REQUESTED** [review]
This looks great! Just one small request and then a quick question that will lead to a second small request regardless of your answer :)

File: cupid/run.py (hunk 1 above)
Comment (nusbaume) [reply cupid/run.py:130]: I am happy with either method of writing the if-statement.  I personally prefer the `if-else` method, but have found that some static analysis tools like `pylint` will complain about it, hence I went the `var=value, if condition: var=other_value` way in this PR.  Do you all have a preference one way or the other?

Comment (mnlevy1981) [reply cupid/run.py:130]: Let's stick with `if-else`, and if we eventually add `pylint` or some other checker that complains we'll just have one more block to update.

Comment (nusbaume) [reply cupid/run.py]: Sounds good!  I've applied the requested changes.

**Review by mnlevy1981 — APPROVED** [review]
looks great!

**Pull request merged**